### DISCUSSION TITLE
Why does the Next App not have the "sideEffects": false, inside packages.json?

### DISCUSSION THREAD

**Discussion by welschmoor** [belgattitude/nextjs-monorepo-example#3383] [General]
Would this not decrease the bundle size of a Next build? I am curious! Thanks!

**Comment by belgattitude** [comment]
Best is to try, but afaik it won't be make any difference in the context of an app. 

The code is not transpiled so webpack/uglify/swc can properly drop unused code. 

That differs in a published library.

**Reply by welschmoor** [reply]
Thank you, belgattitude. Today I just rewrote the barrel files and the result was the same. The shared initial js load (seen during yarn build of next) went from 400kB to 130kB. I'm not sure it makes any difference though, because the number of requests seen under Network Tab of Dev Tools on home page went from slightly above 100 to 156 (the number goes up in both cases by either getting rid of barrels or by using sideEffects: false). I really don't know how to interpret it...

**Reply by belgattitude** [reply]
Hello @welschmoor I'm not sure to follow from side-effects to barrel... that said you might have spotted a nice thing, could you provide a PR / fork where I can have a look to the changes ? Thanks

**Reply by belgattitude** [reply]
@welschmoor Incidentally I've found https://github.com/vercel/next.js/issues/12557 that gives some nuance (external packages, barels). That said sideEffects on a nextjs app does not seem to make any difference.
